pkg/controller/controller.go

```diff
@@ -220,12 +238,24 @@ func New(ctx context.Context, cfg *config.Config, kubeclientset kubernetes.Inter
 		wfLauncher = launchplan.NewFailFastLaunchPlanExecutor()
 	}
 
+	namespaceFilterR := func() *regexp.Regexp {
```

Comment:
Is there a reason for this predicate?

Reply:
I had hard time dealing with `nil`. What I want was:

```
a := nil

if b {
    a = "something not nil"
}
```

It's my bad `go` skill I believe.